### PR TITLE
Feature/tag overview

### LINKED ISSUE #1151: Tag overview in frontend
The frontend should include a page where users can:

- [ ] See a list of their tags
- [ ] Make changes to their tags (such as renaming), or remove tags
- [ ] See in which corpora a tag has been used, and how many documents have been tagged in that corpus
- [ ] Click a link to the tagged results in that corpus

### PR DESCRIPTION
**Pull request by JeltevanBoheemen**
Closes #1151

The table looks a little bulky. Any low-hanging-fruit suggestions for improving the appearance are welcome.

Was contemplating making the tag edit modal a separate component, but decides against it. It reduces some bloat in the tag overview html, but just introduces more `@Input` & `@Output` stuff. Willing to change my mind if you guys would prefer that though. 

To optimally test the component:
- create some tags through the tag overview
- tag some documents in several corpora
- back to the tag overview to test the filtered corpus links

### CODE REVIEW

File: frontend/src/app/tag/tag-overview/tag-overview.component.html [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+</p-table>
+</div>
+</td>
+</tr>
+
+</ng-template>
+
+        </p-table>
+    </ng-container>
+</section>
```
Comment (JeltevanBoheemen) [inline frontend/src/app/tag/tag-overview/tag-overview.component.html:105]: Good job formatter, good job

**Comment by lukavdplas** [comment]
Bug found!

- Go to a corpus
- Open a document and create a new tag for it
- Use the top navigation to navigate to the tags page
- The new tags shows up, but when I unfold it, it doesn't load the corpora

screenshot of tag overview

The corpus will only show up after I reload the page.

**Comment by lukavdplas** [comment]
Also a (very minor) bug, but this is probably a primeNG issue.

When I open and close the details of a tag, some of the borders on the right side of the table disappear.

[screen recording of the tag overview page, showing the effect described](https://github.com/UUDigitalHumanitieslab/I-analyzer/assets/43678097/848a46ee-fbcf-4fe7-8afe-c07ccb0f36fe)

**Comment by lukavdplas** [comment]
> The table looks a little bulky. Any low-hanging-fruit suggestions for improving the appearance are welcome.

Users may not have that many tags, so unlike with the search history, perhaps a table layout isn't necessary?

Here I played around a bit with using cards instead:

screenshot of tag page showing a small panel for each tag rather than a table

That kind of layout may work better if you have a lot of actions and information for each tag.

**Comment by JeltevanBoheemen** [comment]
Bugs were fixed, and switched to cards layout. 
Not 100% happy with the looks, but uninspired, so will hold off on the visual renovation for later.

**Review by lukavdplas — APPROVED** [review]

**Pull request merged**